### PR TITLE
TOhtml fixes

### PR DESCRIPTION
**Pull request by altermo**
I looked through the code and noticed a few bugs and other unwanted behavior:
+ There was an `ipairs()` which should have been a `pairs()`
+ Inlay hints work similarly to semantic tokens, and thus should not be shown
+ When there are multiple notifications, only the last one is shown, so instead add a `(+ %d more warnings)` to the last one.

### CODE REVIEW

File: runtime/lua/tohtml.lua [hunk 1]
```diff
@@ -611,9 +636,7 @@ local function styletable_folds(state)
     end
   end
   if has_folded and type(({ pcall(vim.api.nvim_eval, vim.o.foldtext) })[2]) == 'table' then
-    vim.notify_once(
-      'Info(TOhtml): foldtext returning a table is half supported, HTML may be incorrect'
-    )
+    notify('foldtext returning a table is half supported, HTML may be incorrect')
```
Comment (clason) [inline runtime/lua/tohtml.lua:639]: What does "half supported" mean? Either it is supported, or not ;) (In other words, this message should be consistent with the others.)

Comment (altermo) [reply runtime/lua/tohtml.lua]: notification updated

File: runtime/lua/tohtml.lua [hunk 2]
```diff
@@ -57,6 +57,26 @@
...
+--- @type string[]
+local notifications = {}
+
+---@param msg string
+local function notify(msg)
+  if #notifications == 0 then
+    vim.schedule(function()
+      if #notifications > 1 then
+        vim.notify(
+          ('TOhtml: %s (+ %d more warnings)'):format(notifications[1], tostring(#notifications - 1))
```
Comment (justinmk) [inline runtime/lua/tohtml.lua:69]: Maybe vim.notify should do this by default. Eventually `:echo` and all messages will be reworked to behave something like this (with sufficient affordances for seeing the full message).

**Review by justinmk — APPROVED** [review]

**Review by wookayin — COMMENTED** [review]
LGTM too. My review was too late (started writing before merged), but leaving some nit/minor comment for future recommendations/improvements.

**Pull request merged**